components/tools/OmeroPy/src/omero/util/cleanse.py

```diff
@@ -329,6 +334,56 @@ def fixpyramids(data_dir, query_service,
                         os.remove(pixels_file)
 
 
+def removepyramids(client, little_endian=False, dry_run=False,
+                   imported_after=None, wait=25):
+    client.getImplicitContext().put(omero.constants.GROUP, '-1')
+    admin_service = client.sf.getAdminService()
+    config_service = client.sf.getConfigService()
+
+    initial_check(config_service, admin_service)
+
+    # look for any pyramid files with the specified endianness
+    # the pyramid file will be removed
+    to_delete = []
+    request = omero.cmd.FindPyramids()
+    request.littleEndian = little_endian
+    request.importeAfter = -1
+    if imported_after is not None:
+        date = time.strptime(imported_after, "%d/%m/%Y")
+        request.importeAfter = time.mktime(date)
+    rsp = submit(client, request, wait)
+
+    if len(rsp.pyramidFiles) == 0:
+        print "No pyramids to remove"
+
+    # Prepare the requests
+    for j in range(len(rsp.pyramidFiles)):
+        image_id = rsp.pyramidFiles[j]
+        if dry_run:
+            print "Would remove pyramid for image %s" % image_id
+        else:
+            print "Removing pyramid for image %s" % image_id
+            req = omero.cmd.ManageImageBinaries()
+            req.imageId = image_id
+            req.deletePyramid = True
+            req.deleteThumbnails = True
+            to_delete.append(req)
+
+    if len(to_delete) > 0:
+        submit(client, to_delete, wait)
+        print "%s Pyramids removed" % len(to_delete)
+
+
+def submit(client, request, wait):
+    if isinstance(request, list):
+        request = DoAll(request)
+    ms = 500
+    loops = ceil(wait * 1000.0 / ms)
+    cb = client.submit(request, loops=loops, ms=ms, failonerror=True,
```

Comment:
Here you can either pass `failonerror=False` to get back the `ERR` response or you can catch `omero.CmdError` and unwrap yourself (`cmd_error.err`). Also, the callback will need closing.